### ISSUE TITLE
gnus manual is missing - can't install

### ISSUE THREAD

**Issue by PavoDive** [alexmurray/emacs-snap#75]
This is my setup:

```
user@Machine:~$ lsb_release -a
    No LSB modules are available.
    Distributor ID:	Ubuntu
    Description:	Ubuntu 20.04.6 LTS
    Release:	20.04
    Codename:	focal
user@Machine:~$ which emacs
    /snap/bin/emacs
user@Machine:~$ emacs --version
    GNU Emacs 29.1
    Development version 28fb02492c24 on master branch; build date 2023-09-23.
    Copyright (C) 2023 Free Software Foundation, Inc.
    GNU Emacs comes with ABSOLUTELY NO WARRANTY.
    You may redistribute copies of GNU Emacs
    under the terms of the GNU General Public License.
    For more information about these matters, see the file named COPYING.
```

I wanted to read the `gnus` manual _inside emacs_, but it's apparently missing. As per [one of the answers to this post in emacs.stackexchange](https://emacs.stackexchange.com/questions/32246/wheres-the-gnus-info-manual-on-ubuntu), this may be because of conflicts between the GNU documentation license and the Debian guidelines. The answer suggests installing the `emacs24-common-non-dfsg` (or corresponding version: 26 at the moment of writing) through `apt`, which isn't really viable in my case, as I installed emacs through snap.

This is the screenshot of my emacs info (`C-h i`):

![This is the screenshot of my emacs info (`C-h i`)](https://github.com/alexmurray/emacs-snap/assets/9052269/5c589d35-917f-40fd-9d8c-8fb178f2efe0)

In the "emacs" manual, I can find that gnus is **one of the capters** (chapeter 34, under "advanced features"), but the full manual is missing (the full manual can be found online at https://gnus.org/manual/gnus.html ).

I asked a related question [in askubuntu.com](https://askubuntu.com/questions/1492464/how-to-install-emacs-documentation-if-emacs-was-installed-by-snap) and was directed to file an issue here.

What can be done to install the gnus manual into emacs' snap?

**Comment by alexmurray** [comment]
Thanks for filing this issue but I am unable to reproduce it on a fresh Ubuntu 20.04 VM:

```
sudo snap install --classic emacs
emacs

# then inside emacs
M-x info
# C-s to find Gnus on that page - then can hit enter on the result to open the gnus manual
```

![image](https://github.com/alexmurray/emacs-snap/assets/56540/ddcd8adc-05c2-4e6d-a420-dace9bd818f0)

![image](https://github.com/alexmurray/emacs-snap/assets/56540/2b520890-6bc1-43a9-a2c2-d8e5bb4c6030)

**Comment by alexmurray** [comment]
More details from the test above:

```
amurray@sec-focal-amd64:~$ emacs --version
GNU Emacs 29.1
Development version 28fb02492c24 on master branch; build date 2023-09-23.
Copyright (C) 2023 Free Software Foundation, Inc.
GNU Emacs comes with ABSOLUTELY NO WARRANTY.
You may redistribute copies of GNU Emacs
under the terms of the GNU General Public License.
For more information about these matters, see the file named COPYING.
amurray@sec-focal-amd64:~$ snap info emacs
name:      emacs
summary:   GNU Emacs is the extensible self-documenting text editor
publisher: Alex Murray (alexmurray✪)
store-url: https://snapcraft.io/emacs
contact:   https://github.com/alexmurray/emacs-snap/issues
license:   GPL-3.0+
description: |
  Emacs is the extensible, customizable, self-documenting real-time
  display editor.
  
  Features include:
    * Content-aware editing modes, including syntax coloring, for many file types.
    * Complete built-in documentation, including a tutorial for new users.
    * Full Unicode support for nearly all human scripts.
    * Highly customizable, using Emacs Lisp code or a graphical interface.
    * An entire ecosystem of functionality beyond text editing, including a project
      planner, mail and news reader, debugger interface, calendar, and more.
    * A packaging system for downloading and installing extensions.
  
  This snap is built via the build.snapcraft.io service from the
  snapcraft.yaml definition at https://github.com/alexmurray/emacs-snap to
  ensure source and build transparency.
commands:
  - emacs.ctags
  - emacs.ebrowse
  - emacs
  - emacs.emacsclient
  - emacs.etags
snap-id:      4gE2S31SnK2LeqsZF8XTPuHOlqlzbqsW
tracking:     latest/stable
refresh-date: today at 12:40 ACDT
channels:
  latest/stable:    29.1                   2023-09-23 (2249) 345MB classic
  latest/candidate: 29.1                   2023-09-23 (2249) 345MB classic
  latest/beta:      29.1                   2023-09-23 (2249) 345MB classic
  latest/edge:      30.0.50-master-65600b9 2023-11-21 (2283) 347MB classic
  pgtk/stable:      –                                              
  pgtk/candidate:   –                                              
  pgtk/beta:        –                                              
  pgtk/edge:        30.0.50-master-65600b9 2023-11-21 (2284) 346MB classic
  28.x/stable:      28.2                   2023-09-23 (2192) 241MB classic
  28.x/candidate:   ↑                                              
  28.x/beta:        28.2                   2023-09-23 (2256) 241MB classic
  28.x/edge:        ↑                                              
installed:          29.1                              (2249) 345MB classic
amurray@sec-focal-amd64:~$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 20.04.6 LTS
Release:	20.04
Codename:	focal
```

**Comment by alexmurray** [comment]
It also works fine under a graphical emacs frame too:

![image](https://github.com/alexmurray/emacs-snap/assets/56540/c57e47ca-22a4-402f-80db-08e36bb098d0)

**Comment by PavoDive** [comment]
Hi Alex,
Thanks for taking the time to check this.

I uninstalled emacs completely, reinstalled it with the `--classic` flag, and still see the gnus manual is missing:

```
user@Machine:~$ sudo snap remove emacs
[sudo] password for user: 
emacs removed
user@Machine:~$ emacs --version
bash: emacs: command not found
user@Machine:~$ sudo snap install --classic emacs
emacs 29.1 from Alex Murray (alexmurray✪) installed
user@Machine:~$ emacs
```

This is what I get after `C-h i` in the emacs session:

![image](https://github.com/alexmurray/emacs-snap/assets/9052269/03d33837-e571-4035-9973-7fbd5d9d9cc3)

As you can see, I still have the ESS, Magit, Company and emacs manuals (among others), but the gnus one is missing.

Am I missing any sort of package or library?

Alternatively, is there a way to download the tar.gz file of the manual and manually install it? and if so, where?

**Comment by PavoDive** [comment]
After reading a lot, I found that there is a version-independent `common-non-dfsg` package, so I installed it with `sudo apt-get install emacs-common-non-dfsg` and a lot of manuals (not only gnus) were downloaded to `/usr/share/info`. I also realized that the environment variable `INFOPATH` dind't have `/usr/share/info` among the entries, so I did `INFOPATH = /usr/share/info:$INFOPATH; export INFOPATH`, and now I have the gnus manual (among many others).

Thanks Alex for a great snap and for your help!
